src/integer_mod_q/polynomial_ring_zq/get.rs

```diff
@@ -53,6 +100,83 @@ impl PolynomialRingZq {
     pub fn get_poly(&self) -> PolyOverZ {
         self.poly.clone()
     }
+
+    /// Returns the degree of a [`PolynomialRingZq`] as a [`i64`].
+    /// The zero polynomial has degree '-1'.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer_mod_q::{PolynomialRingZq, ModulusPolynomialRingZq};
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly = PolyOverZ::from_str("3  0 1 1").unwrap();
+    /// let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+    ///
+    /// let degree = poly_ring.get_degree();
+    /// assert_eq!(2, degree);
+    /// ```
+    pub fn get_degree(&self) -> i64 {
+        unsafe { fmpz_poly_degree(&self.poly.poly) }
+    }
+}
+
+#[cfg(test)]
+mod test_get_coeff {
+    use crate::{
+        integer::{PolyOverZ, Z},
+        integer_mod_q::{ModulusPolynomialRingZq, PolynomialRingZq},
+        traits::GetCoefficient,
+    };
+    use std::str::FromStr;
+
+    /// Ensure that 0 is returned if the provided index is not yet set.
+    #[test]
+    fn index_out_of_range() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let poly = PolyOverZ::from_str("4  -1 0 1 1").unwrap();
+        let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+
+        let zero_coeff = poly_ring.get_coeff(4).unwrap();
```

Comment:
Understanding that the 4th coefficient is not set is unnecessary hard. 
Please initialize the polynomial with in a way that it is already reduced.